### ISSUE TITLE
Thank you very much for your UI! Have few suggestion if u would mind.

### ISSUE THREAD

**Issue by alanhk147** [GrandaddyShmax/audiocraft_plus#3]
1) Can we make defoult location to output files like ./output/date/ ? If genegate alot of  files - difficult to navigate.
2) Can we store generation infomation (prompt, settings, sid)  inside sound file? Or if it imposible may be generate small .txt file same name as sound file. Its will be very useful.

**Comment by GrandaddyShmax** [comment]
1. Added to my list
2. already working on it, though not sure if trying to fit the prompt as a name is a good idea, i think ill do them by date, or perhaps ill let the user name it

**Comment by alanhk147** [comment]
I study how to store tag information in wav files. pls lood to my python code.
https://drive.google.com/drive/folders/1FebNTpK4TM3qxbIQWfEt29X5TV_lWGJS?usp=sharing

**Comment by GrandaddyShmax** [comment]
Sorry for the late reply, i am now trying your method, earlier tried to accomplish it with a different method but it only brought more issues. Do you know if this way i can also give tags to the mp4 format? or does it only work with audio files like 'wav'?

**Comment by alanhk147** [comment]
Hi !  I testd for .mp3 and .wav. Its shold work with mp4 too. Let  me test and reply you.  Bur frendly speaking .mp4 from my point its wrong way to store music. mp4 comression kill 40% of quality. Out system not genegare video - whe use mp4? Not sure about you but from my point better cancel mp4 output, or at least make it optional. 

Actually  here is my email: alanhk147@gmail.com

**Comment by alanhk147** [comment]
Yes its working with .pm4 too. Just replace in my code to  file_path = r'.\test.mp4' and put to code folder any file with test.mp4 name. 
here is existing tags one of generated by MGplus m4 files:
{'ENCODEDBY': ['Lavf60.3.100']}

**Comment by alanhk147** [comment]
here is described limitations for TAG sizes. Looks like all our information can easy be stored.
https://mutagen-specs.readthedocs.io/en/latest/id3/id3v2.4.0-structure.html

**Comment by GrandaddyShmax** [comment]
> Yes. My code is working with .mp4 too. Just change in my code file_path = r'.\test.mp4' and put to code folder any file with test.mp4 name. here is existing tags one of generated by MGplus mp4 files: {'ENCODEDBY': ['Lavf60.3.100']}

After playing around with taglib, when i give it custom tags, it saves them and shows them up when i print them. but when i upload that same file back it only shows the default tag as if it did not save anything

**Comment by GrandaddyShmax** [comment]
here is an example, here it shows the tags before and after modifying them:
![image](https://github.com/GrandaddyShmax/audiocraft_plus/assets/52707645/2cc94c58-47b9-40db-84da-316ed2b18829)

but when i reupload them it only shows me this:
![image](https://github.com/GrandaddyShmax/audiocraft_plus/assets/52707645/a8a13def-02ab-405d-b476-a04f0e2aad33)

**Comment by alanhk147** [comment]
I am 100% sure it’s can stored. Today I be outside but tomorrow I will make for you other example :)

**Comment by GrandaddyShmax** [comment]
show me how you exported the mp4 file and then how you uploaded it and read the tags.
alternatively if you already know what to do, you can modify the app.py and do a pull request to the test branch. in addition ill add you as a contributor of this webui

**Comment by alanhk147** [comment]
Haaa :) thank you very much to trust. Actually I am very interesting to develop this project. One of my hobby region for many years already - music.  B4 I not touch web GUI - prefer local QT GUI but I do not think it’s too difficult to study. Any way I will be happy to joint your project.  Let’s communicate tomorrow when I get back to computer :)

**Comment by alanhk147** [comment]
Hi !
https://drive.google.com/drive/folders/1_xUUOMi0wsTuiipPBLioWQ9cE5edxXH8?usp=sharing
Here is 3 files. 
1) metadata_src.wav original file with some MUSIC tags in it.
2)  wav_Tag_reader.py - code to read existing tags for 2 files metadata_src.wav and metadata_dst.wav (if second one exist)
3) Wav_Tag_Changer.py - code to read metadata_src.wav file, pring its tags, then make its copy  to metadata_dst.wav, modify TAGS to that we need , then reload and print modified file tags 

for App.py i need time to study.  unfortunately looks like its imposible to debug in standart IDE ( I am using WIng Pro). So until now i do not understand where exactly genegared files wrings to hard drive. If u help me explain that - its will be faster for me to make App file modifications by myself.

**Comment by alanhk147** [comment]
for .mp4 situation little more complicated. There is only few avalible tag names. If try to use diferent names - TAG become empty.But inside every tag - there is list. So can encode any information in it.
just add to folder two more files
https://drive.google.com/drive/folders/1_xUUOMi0wsTuiipPBLioWQ9cE5edxXH8?usp=sharing

4) metadata_scr.mp4  example of some generated output file. 
5) mp4_Changer.py - code for make copy and modify avalible tags name in copy file.

**Comment by GrandaddyShmax** [comment]
sorry, i've been busy these last days, i will now check them out 👍

**Comment by alanhk147** [comment]
Its ok. already understand Gradio logic little. Tomorrow will provide u modified version of app.py ;)

**Comment by GrandaddyShmax** [comment]
are you sure that you checked the mp4 file after exporting and importing again? because when i upload the file that you provided, it shows me this:
![image](https://github.com/GrandaddyShmax/audiocraft_plus/assets/52707645/f6da1f84-1f4a-418b-b134-c9d5674df3ba)

and this is the function that handles with printing the text from the mp4, just as in your example:
![image](https://github.com/GrandaddyShmax/audiocraft_plus/assets/52707645/0f51c35b-e9af-450f-b457-444cbb8312af)

perhaps it's something to do with gradio, maybe it strips the file from it's metadata 🤷‍♂️

**Comment by alanhk147** [comment]
https://drive.google.com/drive/folders/1XDD5niUhBHjCR9NDxes6uBcPTTWj81PQ?usp=sharing
here is app.py and folder with 2 generated files.  I just not made yet procedure to convert real reneration paraneters to list that will stored in mediadta insted I just for time been generate example list. 
what i did 
Add 3 new libs in row 10

Add 4 new procedures start from row 300
def convert_tags(inp):
def add_tags(filename,tags): 
def save_outputs(mp4, wav_tmp, seed):
def clear_cash():

arr 2 lines to predict_full procedure at row 425 and row 480

my Telegramm ID is: alanhk147 

;)

**Comment by GrandaddyShmax** [comment]
huh, im still not sure why in your version it works but it just does, amazing work! 🔥
I will edit it to work with the latest changes

btw i do not have telegram so if you have a discord account, send me it

**Comment by GrandaddyShmax** [comment]
It starts to look a lot nicer 😊
![image](https://github.com/GrandaddyShmax/audiocraft_plus/assets/52707645/85ff4389-a46f-4d33-9d37-3fa6f3da7e90)

**Comment by alanhk147** [comment]
my discord username is: alanhk#9061

**Comment by alanhk147** [comment]
i mede some pseudo stereo improvement in stereo effect mode. Actually from my point "stereo" mode better rename to "2ch mono" coz its not related to stereo at all - just copy to stereo channels same mono chennel.  
row 14 : import torchaudio
row 177:  def make_pseudo_stereo (filename, sr_select, pan,delay): #combine your and my methods in one procedure
row 286:  replace your pseudo stereo method code by call of procedure 

https://drive.google.com/drive/folders/1N7bKAad40-utY7-gXhM6ipBWcETOTLrD?usp=sharing

**Comment by GrandaddyShmax** [comment]
> my discord username is: alanhk#9061

ive sent you a friend request



> i made some pseudo stereo improvement in stereo effect mode. Actually from my point "stereo" mode better rename to "2ch mono" coz its not related to stereo at all - just copy to stereo channels same mono chennel. row 14 : import torchaudio row 177: def make_pseudo_stereo (filename, sr_select, pan,delay): #combine your and my methods in one procedure row 286: replace your pseudo stereo method code by call of procedure
> 
> https://drive.google.com/drive/folders/1N7bKAad40-utY7-gXhM6ipBWcETOTLrD?usp=sharing

it does sound slightly better actually, i see you merged the mono with stereo, it's really good, i've added it as well.
Thanks 😊
